.mvn/gradle-enterprise.xml

```diff
@@ -22,7 +22,9 @@
 <gradleEnterprise
         xmlns="https://www.gradle.com/gradle-enterprise-maven" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="https://www.gradle.com/gradle-enterprise-maven https://www.gradle.com/schema/gradle-enterprise-maven.xsd">
-  <enabled>#{env['GRADLE_ENTERPRISE_ACCESS_KEY']?.trim() > ''}</enabled>
+  <!-- Enable Gradle Enterprise extension when GRADLE_ENTERPRISE_ACCESS_KEY is set and the build isn't
+       a pull request from a branch or forked repository with a name that indicates it's a work in progress. -->
+  <enabled>#{env['GRADLE_ENTERPRISE_ACCESS_KEY']?.trim() > '' and !(env['GITHUB_HEAD_REF']?.matches('(?i).*(experiment|wip|private).*') or env['GITHUB_REPOSITORY']?.matches('(?i).*(experiment|wip|private).*'))}</enabled>
```

Comment:
If GITHUB_HEAD_REF is null, then `env['GITHUB_HEAD_REF']?.matches` will return false making the `<enabled>`, enabled?

Reply:
no, there is `!` and parenthesis around a `OR` condition between the 2 rules. something like  `!(<GITHUB_HEAD_REF check> OR <GITHUB_REPOSITORY check>)`